src/driver/DriverUtils.ts

```diff
@@ -61,8 +61,8 @@ export class DriverUtils {
         return {
             type: type,
             host: host,
-            username: username,
-            password: password,
+            username: decodeURIComponent(username),
+            password: decodeURIComponent(password),
```

Comment:
are you sue other parameters like host shouldn't be decoded as well?

Reply:
In theory, we should decode every part that contains "reserved" characters (cf. https://en.wikipedia.org/wiki/Percent-encoding). In practice, in the 15 years of my career, I haven't see any hostname or database name using reserved characters.